client/src/parser.js

```diff
@@ -0,0 +1,99 @@
+const { parse } = require('path');
+
+/**
+ * @description Parses an JSON created from the .bpmn xml of the model to an executable .robot file
+ * @returns {string} Code that has to be put in .robot file
+ */
+
+// this should be refactored to a dictionary
+function taskLib(app) {
+  if (app == 'Excel.Files') return 'RPA.Excel.Files';
+  if (app == 'Browser') return 'RPA.Browser';
+}
+
+function parseDiagramJson(json_data) {
+  let parsedCode = '';
+  var bpmnTasks = json_data['bpmn2:definitions']['bpmn2:process']['bpmn2:task'];
+
+  // create Settings statements
+  parsedCode += '*** Settings ***\n';
+  parsedCode += 'Documentation  Our first parsed RPA\n';
+
+  if (bpmnTasks !== undefined) {
+    console.log('bpmnTasks: ' + bpmnTasks);
+    let codeToAppend = '';
+    if (typeof bpmnTasks[0] === 'object') {
+      // collect apps being used
+      const applications = [];
+      for (let task in bpmnTasks) {
+        const rpaApp =
+          bpmnTasks[task]['bpmn2:extensionElements']['camunda:properties'][
+            'camunda:property'
+          ][0]._attributes.value;
+        if (!applications.includes(rpaApp)) {
+          applications.push(rpaApp);
+        }
+      }
+      console.log(applications);
+
+      // import libraries needed for apps
```

Comment:
Code should document itself, remove the comment and maybe create a new function to be called with it as the name or shift the comment into the functions description

Reply:
Also the same for the following comments for smaller sections